stand/efi/loader/bootinfo.c

```diff
@@ -433,7 +438,7 @@ bi_load(char *args, vm_offset_t *modulep, vm_offset_t *kernendp, bool exit_bs)
 	module = *modulep;
 	file_addmetadata(kfp, MODINFOMD_MODULEP, sizeof(module), &module);
 #endif
-#ifdef EFI
+#if defined(EFI) && !defined(__i386__)
```

Comment:
Why omit this on x86?

Reply:
Some of its members are `UINTN`s, which is 8 bytes on amd64 and 4 bytes on x86. Even if the kernel was modified to detect this, all the efirt functions are required to be run in protected mode.